src/realm/table.hpp

```diff
@@ -1641,6 +1643,19 @@ inline bool Table::has_shared_type() const noexcept
     return !m_top.is_attached();
 }
 
+inline void Table::verify_column(size_t col_ndx, const ColumnBase* col) const
+{
+    if (col_ndx < m_cols.size()) {
+        if (m_cols[col_ndx] == col)
+            return;
+        // The column might be elsewhere in the list
+        for (auto it : m_cols) {
+            if (it == col)
+                return;
+        }
```

Comment:
@simonask it is not luck ;-). When c++ allows `if (col in m_cols)`, I will happily change to that. 'it` renamed.